pkg/api/types.go

```diff
@@ -448,6 +448,8 @@ const (
 // Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
 	Path string `json:"path"`
+	// Mount the host path with propagation shared. Docker only.
+	Shared bool `json:"shared"`
```

Comment:
I do NOT want to expose the full complexity of the linux kernel here.  I'd love to keep it to a bool, and let the implementation decide what that means (I forget the slave semantics).